src/scene/container/utils/assignPropertiesFromOptions.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Assigns properties from one object to another, using an optional array of property names to ignore.
+ * @param target - The target object to assign properties to.
+ * @param options - The object to assign properties from.
+ * @param ignore - An object of property names to ignore ({ propToIgnore: true }).
+ */
+export function assignPropertiesFromOptions<T extends Record<string, any>>(
+    target: T,
+    options: T,
+    ignore: Record<string, boolean> = {}
```

Comment:
Why not `string[]` instead?

If not, the type should probably be `Record<string, true>`.